library/std/math.qs

```diff
@@ -45,6 +45,24 @@ namespace Microsoft.Quantum.Math {
         0.6931471805599453
     }
 
+    //
+    // Special numbers in IEEE floating-point representation
+    //
+
+    /// # Summary
+    /// Returns whether a given floating-point value is not a number (i.e. is
+    /// NaN).
+    ///
+    /// # Input
+    /// ## d
+    /// A floating-point value to be checked.
+    ///
+    /// # Output
+    /// `true` if and only if `d` is not a number.
+    function IsNaN(d : Double) : Bool {
```

Comment:
Could you add a test case to https://github.com/microsoft/qsharp/blob/d183b05a863d8f2365f61392839d3fd703eedb46/library/tests/src/test_math.rs to cover this new function?

Reply:
Added test case.